### ISSUE TITLE
OG descriptions and twitter descriptions are too big

### ISSUE THREAD

**Issue by michaeltorbert** [semperfiwebdesign/all-in-one-seo-pack#808]
With "Use Content For Autogenerated OG Descriptions" and "Autogenerate Meta Descriptions" unchecked, no character count filter is applied, and the length of the descriptions is too much. (This may also apply to other settings as well.)

I've verified the issue.

I believe this was introduced in 2.3.12, but we should confirm this.

Reported by Barry via email.

**Comment by arnaudbroes** [comment]
On it.

**Comment by michaeltorbert** [comment]
It looks like the OG desc and twitter desc are coming in at ~1000 characters max.
According to their docs, Facebook can do up to 300 and Twitter can do up to 200.

**Comment by arnaudbroes** [comment]
@michaeltorbert 
Tested with about 8 versions, I can confirm this isn't a new.
.

**Comment by contactashish13** [comment]
@michaeltorbert this seems to be a problem with other settings as well. While testing, I tried a long title and the entire title was printed out without any truncation. Do you want to create another issue to deal with this or should this done as part of the same issue?

**Comment by michaeltorbert** [comment]
@contactashish13 Since that's a separate issue, it's best to go ahead and create a separate Github issue.

**Comment by contactashish13** [comment]
### Facebook

I've been searching about this and it seems there are multiple schools of thought without a consensus:

1. https://stackoverflow.com/a/8916327 says provide whatever and it will be suitably truncated
1. The official site http://ogp.me/ says very vaguely "A one to two sentence description of your object" without a hard limit
1. https://stackoverflow.com/a/9552296 (same post) refers to a link that is dead but the content from there is what shows up when you google "og:description length"
1. https://stackoverflow.com/questions/8914476/facebook-open-graph-meta-tags-maximum-content-length#comment76342330_8916327 (same post) says there is a 10,000 character limit which seems to be something we don't cross.

I'm not sure we should be imposing any limits here.

### Twitter 
On the other hand, Twitter is explicit https://developer.twitter.com/en/docs/tweets/optimize-with-cards/overview/markup.html

### Conclusion
I think we should create a separate issue for twitter meta tags limits and address them all in one issue. For Facebook, we should create a separate issue and conclude there, if we can.

suggestions @arnaudbroes @wpsmort @wpsmort?

**Comment by arnaudbroes** [comment]
I suggest that we use the optimal OG lengths I brought up in #2139. Wait on what Michael and Steve have to say about this though.

**Comment by wpsmort** [comment]
Here's Facebook's documentation:

https://developers.facebook.com/docs/sharing/best-practices/#tags
https://developers.facebook.com/docs/sharing/webmasters#markup

I've checked several popular pages on Facebook and they only seem to display one line of the description.  I therefore agree with Arnaud that we should truncate the description based on the length in #2139.

**Comment by arnaudbroes** [comment]
@contactashish13 I'll just post those lengths here again for your convenience.

- OG Title length: 55 characters or under.
- OG Description length: 55 characters.

Just make sure there's no truncation.

**Comment by contactashish13** [comment]
@wpsmort Just being the devil's advocate here, why are we going against Facebook's cague documentation and instead going with arbitrary lengths. The lengths Arnaud is recommending is probably from https://www.thoughtshift.co.uk/pimp-my-serp-are-open-graph-og-tags-more-important-than-meta-data/. Do we have official data to back this up? 

@arnaudbroes what do you mean no truncation?

**Comment by wpsmort** [comment]
We agreed in the dev meeting today that, in the absence of any official documentation from Facebook, we'll go with the information in the thoughtshift.co.uk article that Arnaud found and we'll make it clear in our documentation and help that this is just a guide

**Comment by contactashish13** [comment]
PR https://github.com/semperfiwebdesign/all-in-one-seo-pack/pull/2297
